### DISCUSSION TITLE
Wireable serialization should take precedence over Eloquent serialization

### DISCUSSION THREAD

**Discussion by mabdullahsari** [livewire/livewire#3839] [7. Bugs (archived)] [answered]
Unless this is the intended behaviour, the `Wireable` contract should be respected even if it is applied on an Eloquent model.

I don't see anything being mentioned in the [original PR](https://github.com/livewire/livewire/pull/3362), so I'm assuming this is an oversight.

**Comment by joshhanley** [comment, marked as answer]
@mabdullahsari yeah agreed, I don't think it was intentional. Can you submit a PR?

Thanks!

**Reply by mabdullahsari** [reply]
Sure thing.

https://github.com/livewire/livewire/pull/3856

**Reply by joshhanley** [reply]
Great thanks! 🙂

**Comment by riesjart** [comment]
@mabdullahsari With this change I guess the order within `HydratePublicProperties::hydrate()` should be changed as well, what do you think?

**Reply by mabdullahsari** [reply]
No, that's not needed. 

`models` will not contain anything when dehydration takes place because it will now be in the `wireables` array.
